components/cli/ballerina/bal_executor.go

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright (c) 2019 WSO2 Inc. (http:www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http:www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package ballerina
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/cellery-io/sdk/components/cli/pkg/constants"
+	"github.com/cellery-io/sdk/components/cli/pkg/util"
+)
+
+type BalExecutor interface {
+	Build(fileName string, iName []byte) error
+}
+
+type LocalBalExecutor struct {
+	execPath string
+}
+
+// NewLocalBalExecutor returns a LocalBalExecutor instance.
+func NewLocalBalExecutor(execPath string) *LocalBalExecutor {
+	balExecutor := &LocalBalExecutor{}
+	balExecutor.execPath = execPath
+	return balExecutor
+}
+
+// Build executes ballerina build on an executable bal file.
+func (baleExecutor *LocalBalExecutor) Build(fileName string, iName []byte) error {
+	cmd := &exec.Cmd{}
+	exePath, err := GetBallerinaExecutablePath()
+	if err != nil {
+		return fmt.Errorf("failed to get executable path, %v", err)
+	}
+	cmd = exec.Command(exePath, "run", fileName, "build", string(iName), "{}")
+	executeBuildCommand(cmd)
+	_ = os.Remove(fileName)
+	return nil
+}
+
+// GetBallerinaExecutablePath returns the ballerina executable path.
+func GetBallerinaExecutablePath() (string, error) {
```

Comment:
Do we need to make this public?

Reply:
Renamed to "ExecutablePath" and made it an interface function.